shared/teams/team/settings-tab/index.js

```diff
@@ -180,35 +180,37 @@ const IgnoreAccessRequests = (props: SettingProps) =>
 export class Settings extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props)
-    this.state = {
-      // This isn't strictly necessary but flow doesn't understand getDerivedStateFromProps will run
-      newIgnoreAccessRequests: this.props.ignoreAccessRequests,
-      newPublicityAnyMember: this.props.publicityAnyMember,
-      newPublicityMember: this.props.publicityMember,
-      newPublicityTeam: this.props.publicityTeam,
-      newOpenTeam: this.props.openTeam,
-      newOpenTeamRole: this.props.openTeamRole,
+    this.state = this._getNewStateObject(props)
+  }
+
+  _getNewStateObject = (p: Props) => {
+    return {
+      newIgnoreAccessRequests: p.ignoreAccessRequests,
+      newOpenTeam: p.openTeam,
+      newOpenTeamRole: p.openTeamRole,
+      newPublicityAnyMember: p.publicityAnyMember,
+      newPublicityMember: p.publicityMember,
+      newPublicityTeam: p.publicityTeam,
       newRetentionPolicy: retentionPolicies.policyRetain, // placeholder
       publicitySettingsChanged: false,
       retentionPolicyChanged: false,
       retentionPolicyDecreased: false,
     }
   }
 
-  static getDerivedStateFromProps(nextProps: Props, prevState: State) {
-    // We just got new settings for this team, reset any user selections
-    // to reflect the actual settings.
-    return {
```

Comment:
this is explicitly called out as being incorrect behavior in the react blog post, we can't just always return a new state w/o looking at the input at all